models/tests/test_gan.py

```diff
@@ -0,0 +1,52 @@
+from pathlib import Path
+import os
+import time
+
+from models.gan.gan_2D import GAN
+
+all_notes = '../midi/tests/test_files/test_notes/test_all_notes.mid'
+sample_file = '../midi/tests/test_files/test_polyphony/test_tempos_velocities_and_polyphony.mid'
+one_note = '../midi/tests/test_files/test_notes/one_note.mid'
+
+
+def test_dataset_creating():
+    model = GAN()
+    model.prepare_data(all_notes)
+    assert len(model.data) >= 1
+
+
+def test_model_loading(tmpdir):
+    if not os.path.exists(tmpdir):
+        os.makedirs(tmpdir, exist_ok=True)
```

Comment:
`.save` fails because of this: You are making it a directory. 

`save`/`load` in gan should be left as they were, otherwise you are now saving in a different path than specified.

To fix tests you can create the directory the following way:

```python
path = Path(tmpdir)
path.mkdir(exist_ok=True)

path = path.joinpath('model')
```

Then save/load from `path`